dist/gcc-compatible/Hacl_MAC_Poly1305.h

```diff
@@ -35,43 +35,46 @@ extern "C" {
 #include "krml/lowstar_endianness.h"
 #include "krml/internal/target.h"
 
-#include "Hacl_Poly1305_32.h"
+#include "Hacl_Krmllib.h"
 
-typedef struct Hacl_Streaming_Poly1305_32_poly1305_32_state_s
+typedef uint64_t *Hacl_MAC_Poly1305_poly1305_ctx;
+
+void Hacl_MAC_Poly1305_poly1305_init(uint64_t *ctx, uint8_t *key);
+
+void Hacl_MAC_Poly1305_poly1305_update1(uint64_t *ctx, uint8_t *text);
+
+void Hacl_MAC_Poly1305_poly1305_update(uint64_t *ctx, uint32_t len, uint8_t *text);
```

Comment:
Wasn't the length arugment supposed to come ofter the data? Or what's the `len` here?

Reply:
yeah same issue the order of arguments is incorrect

Reply:
But what do these functions do? `Hacl_MAC_Poly1305_poly1305_update1` isn't used anywhere and `Hacl_MAC_Poly1305_poly1305_update` seems to me like an auxiliary function used in `Hacl_MAC_Poly1305_mac` and `Hacl_MAC_Poly1305_update`.
In both cases, the logical name for them would be `Hacl_MAC_Poly1305_update` which is redundant with the streaming API.

Reply:
ah! I confused those with the streaming API. I think this module should undergo the same treatment: hide Hacl.Poly1305 underneath Hacl.Streaming.Poly1305